File: backend/WarehouseOps.Domain/Shipment.cs
namespace WarehouseOps.Domain;

public class Shipment : BaseEntity
{
    public Guid OrderId { get; set; }

    public ShipmentStatus Status { get; set; }

    public string TrackingNumber { get; set; } = string.Empty;

    public DateTime? ShippedDate { get; set; }

    public DateTime? DeliveredDate { get; set; }

    public Order? Order { get; set; }
} 

File: frontend/warehouseops-client/src/types/shipment.ts
export type Shipment = {
  id: string;
  orderId: string;
  orderStatus: string;
  customerName: string;
  status: string;
  trackingNumber: string;
  shippedDate: string | null;
  deliveredDate: string | null;
  createdAt: string;
  updatedAt: string | null;
};

export type UpdateShipmentStatusRequest = {
  status: string;
};
